cpp/include/cudf/scalar/scalar_device_view.cuh

```diff
@@ -190,22 +165,22 @@ class timestamp_scalar_device_view
  * @brief Get the device view of a numeric_scalar
  */
 template <typename T>
-auto get_scalar_device_view(numeric_scalar<T>& s) {
+auto get_scalar_device_view(numeric_scalar<T> const& s) {
   return numeric_scalar_device_view<T>(s.type(), s.data(), s.validity_data());
 }
 
 /**
  * @brief Get the device view of a string_scalar
  */
-auto get_scalar_device_view(string_scalar& s) {
+inline auto get_scalar_device_view(string_scalar const& s) {
```

Comment:
Made this inline.  The other functions are templated, this one becomes problematic if you include scalar_device_view in multiple files.